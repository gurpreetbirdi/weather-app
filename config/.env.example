API_BASE_URL="https://example"
APP_ID="your app id for openweathermap"