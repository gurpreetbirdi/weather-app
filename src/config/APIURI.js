// eslint-disable-next-line no-console
console.info('env:', process.env.NODE_ENV);

export const { APP_ID } = process.env;

export const { API_BASE_URL } = process.env;

// end points
export const END_POINTS = {
  GET_WEATHER_INFO: '/data/2.5/forecast?q={LOCATION}&APPID={APP_ID}&units={UNITS}&cnt=40',
};
