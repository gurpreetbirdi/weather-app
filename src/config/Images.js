// Images Path
import applicationErrorImage from '../images/applicationErrorIcon.png';

export const images = { applicationErrorImage };
