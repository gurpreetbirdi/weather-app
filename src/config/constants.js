// error messages
export const errorMsg = {};

// info messages
export const infoMsg = {};

// success messages
export const successMsg = {};

export const constants = {
  CELCIUS: 'metric',
  FAHRENHEIT: 'imperial',
  DATE_FORMAT: 'DD MMM YY', // 01 Jun 20
  TIME_FORMAT: 'hh A', // 09 PM
};
