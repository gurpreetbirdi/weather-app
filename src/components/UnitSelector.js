import React from 'react';
import PropTypes from 'prop-types';
import { Grid, FormControl, RadioGroup, FormControlLabel, Radio } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { constants } from '../config/constants';

const useStyles = makeStyles(() => ({
  grid: {
    height: '10%',
    justifyContent: 'center',
  },
}));

const UnitSelector = ({ unit, handleChange }) => {
  const classes = useStyles();

  return (
    <Grid item container xs={12} className={classes.grid}>
      <FormControl component="fieldset">
        <RadioGroup aria-label="unit" name="unit" row value={unit} onChange={handleChange}>
          <FormControlLabel value={constants.CELCIUS} control={<Radio />} label="Celcius" />
          <FormControlLabel value={constants.FAHRENHEIT} control={<Radio />} label="Fahrenheit" />
        </RadioGroup>
      </FormControl>
    </Grid>
  );
};

UnitSelector.propTypes = {
  unit: PropTypes.string,
  handleChange: PropTypes.func,
};

export default UnitSelector;
