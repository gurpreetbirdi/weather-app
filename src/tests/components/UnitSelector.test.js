import React from 'react';
import { shallow } from 'enzyme';

import UnitSelector from '../../components/UnitSelector';
import { constants } from '../../config/constants';

describe('<UnitSelector />', () => {
  const handleChange = jest.fn();
  const wrapper = shallow(<UnitSelector unit={constants.FAHRENHEIT} handleChange={handleChange} />);

  test('should render UnitSelector correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call handleChange on change', () => {
    wrapper.find('#unit').simulate('change');
    expect(handleChange).toHaveBeenCalled();
  });
});
