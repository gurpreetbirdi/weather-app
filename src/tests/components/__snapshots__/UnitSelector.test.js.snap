// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnitSelector /> should render UnitSelector correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-grid-1"
  container={true}
  item={true}
  xs={12}
>
  <WithStyles(ForwardRef(FormControl))
    component="fieldset"
  >
    <ForwardRef(RadioGroup)
      id="unit"
      name="unit"
      onChange={[MockFunction]}
      row={true}
      value="imperial"
    >
      <WithStyles(ForwardRef(FormControlLabel))
        control={<WithStyles(ForwardRef(Radio)) />}
        label="Celcius"
        value="metric"
      />
      <WithStyles(ForwardRef(FormControlLabel))
        control={<WithStyles(ForwardRef(Radio)) />}
        label="Fahrenheit"
        value="imperial"
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(Grid))>
`;
